ppb_vector/vector2.py

```diff
@@ -98,18 +107,19 @@ def __new__(cls, *args, **kwargs):
 
         if kwargs:
             x, y = kwargs['x'], kwargs['y']
+
         elif len(args) == 1:
             value = args[0]
-            if isinstance(value, cls):
-                # Short circuit if already a valid instance
-                return value
+            # Early return if the argument is a vector:
+            #  __new__ returned the same vector, no need to (re)set x and y
+            if isinstance(args[0], type(self)):
```

Comment:
You called me out once for using `args[0]` instead of `value`

Reply:
And you are entirely right to do that here.  I think it's because it got copy-pasted from `__new__`, where there's no variable `value`.

Reply:
Oh, this came from #146, so I fixed it there too.